File: frontend/package-app.js
const fs = require('fs');
const path = require('path');

const electronPath = path.dirname(require.resolve('electron'));
const distDir = path.join(__dirname, 'dist');
const backendDir = path.join(__dirname, 'backend', 'desktopclaw');
const appDir = path.join(__dirname, 'app');

console.log('Cleaning existing app directory...');
if (fs.existsSync(appDir)) {
    fs.rmSync(appDir, { recursive: true, force: true });
}

console.log('Creating app directory...');
fs.mkdirSync(appDir, { recursive: true });

console.log('Copying Electron...');
const electronDist = path.join(electronPath, 'dist');
fs.cpSync(electronDist, path.join(appDir, 'electron'), { recursive: true });

console.log('Copying frontend dist...');
const appResourcesDir = path.join(appDir, 'resources', 'app');
fs.mkdirSync(appResourcesDir, { recursive: true });

const distFiles = fs.readdirSync(distDir);
for (const file of distFiles) {
    const srcPath = path.join(distDir, file);
    const destPath = path.join(appResourcesDir, file);
    if (fs.statSync(srcPath).isDirectory()) {
        fs.cpSync(srcPath, destPath, { recursive: true });
    } else {
        fs.copyFileSync(srcPath, destPath);
    }
}

console.log('Copying backend...');
fs.cpSync(backendDir, path.join(appDir, 'backend'), { recursive: true });

console.log('Creating start script...');
const startScript = `@echo off
cd /d "%~dp0"
start /B backend\\desktopclaw.exe
timeout /t 5 /nobreak >nul
electron\\electron.exe resources\\app
`;
fs.writeFileSync(path.join(appDir, 'start.bat'), startScript);

console.log('Creating package.json for Electron...');
const packageJson = {
    name: 'desktopclaw',
    version: '1.0.0',
    main: 'main.js'
};
fs.writeFileSync(path.join(appResourcesDir, 'package.json'), JSON.stringify(packageJson));

console.log('Creating main.js for Electron...');
const mainJs = `const { app, BrowserWindow } = require('electron')
const path = require('path')

let win

const createWindow = () => {
    const { width: screenWidth, height: screenHeight } = require('electron').screen.getPrimaryDisplay().workAreaSize
    win = new BrowserWindow({
        width: screenWidth,
        height: screenHeight,
        x: 0,
        y: 0,
        transparent: true,
        frame: false,
        alwaysOnTop: true,
        resizable: false,
        skipTaskbar: true,
        autoHideMenuBar: true,
        webPreferences: {
            devTools: false,
            contextIsolation: true,
            nodeIntegration: false
        }
    })

    win.loadFile('index.html')

    win.webContents.session.webRequest.onHeadersReceived((details, callback) => {
        callback({
            responseHeaders: {
                ...details.responseHeaders,
                'Content-Security-Policy': ["default-src 'self'; script-src 'self' 'unsafe-inline' 'unsafe-eval'; img-src 'self' data: blob:; style-src 'self' 'unsafe-inline'; media-src 'self' blob: http://127.0.0.1:3000 https://127.0.0.1:3000; connect-src 'self' blob: http://127.0.0.1:3000 https://127.0.0.1:3000;"]
            }
        })
    })
}

app.on('ready', createWindow)

app.on('window-all-closed', () => {
    if (process.platform !== 'darwin') app.quit()
})

app.on('activate', () => {
    if (BrowserWindow.getAllWindows().length === 0) createWindow()
})
`;
fs.writeFileSync(path.join(appResourcesDir, 'main.js'), mainJs);

console.log('App packaged successfully!');
console.log(`App location: ${appDir}`);

File: package_app.py
import os
import shutil
import subprocess
from pathlib import Path

def main():
    frontend_dir = Path('frontend')
    dist_dir = frontend_dir / 'dist'
    app_dir = frontend_dir / 'app'
    
    print("Creating app directory structure...")
    
    if app_dir.exists():
        shutil.rmtree(app_dir)
    
    app_dir.mkdir(parents=True)
    
    print("Copying Electron resources...")
    subprocess.run([
        'npx', 'electron', '--version'
    ], check=True, capture_output=True)
    
    import electron
    electron_path = Path(electron.__file__).parent.parent
    
    print(f"Electron path: {electron_path}")
    
    shutil.copytree(
        electron_path / 'dist',
        app_dir / 'electron',
        dirs_exist_ok=True
    )
    
    print("Copying frontend dist...")
    shutil.copytree(
        dist_dir,
        app_dir / 'resources' / 'app',
        dirs_exist_ok=True
    )
    
    print("Copying backend...")
    shutil.copytree(
        frontend_dir / 'backend' / 'desktopclaw',
        app_dir / 'backend',
        dirs_exist_ok=True
    )
    
    print("Creating desktopclaw.exe launcher...")
    with open(app_dir / 'desktopclaw.bat', 'w') as f:
        f.write('@echo off\n')
        f.write('cd /d "%~dp0"\n')
        f.write('start /B backend\\desktopclaw.exe\n')
        f.write('timeout /t 5 /nobreak >nul\n')
        f.write('electron\\electron.exe resources\\app\n')
    
    print("Creating shortcut...")
    import winshell
    from win32com.client import Dispatch
    
    shell = Dispatch('WScript.Shell')
    shortcut = shell.CreateShortCut(str(app_dir / 'desktopclaw.lnk'))
    shortcut.TargetPath = str(app_dir / 'desktopclaw.bat')
    shortcut.WorkingDirectory = str(app_dir)
    shortcut.save()
    
    print("App packaged successfully!")
    print(f"App location: {app_dir}")

if __name__ == '__main__':
    main()
